### ISSUE TITLE
Question on '__fix_for_ie_crossdomain__'

### ISSUE THREAD

**Issue by jihoonkim87** [danialfarid/angular-file-upload#534]
Hello there, 

Somehow I'm getting 404 on __fix_for_ie_crossdomain__, even I got 200 for FileAPI.flash.swf and FileAPI.min.js.

Do you know what it would be? I'm using 2.2.2 by the way.

Thanks a lot.

**Comment by danialfarid** [comment]
I don't think that's related to this plugin or file api. It might be from your other dependencies.

**Comment by rvanbaalen** [comment]
Not sure what this guys question was but @danialfarid I seem to have a similar problem.
When I execute a CORS call from http://sub.domain.com to http://api.domain.com in Internet Explorer 8/9 I too get the `fix_for_ie_crossdomain` response with a 404. In every other browser, the xhr patch is not applied and this problem doesn't exist.

My problem with this is that you seem to be patching xhr for the whole application. All calls go through your xhr patch instead of only upload related calls.

**Comment by rvanbaalen** [comment]
See https://github.com/danialfarid/angular-file-upload/search?utf8=%E2%9C%93&q=_fix_for_ie_crossdomain__

**Comment by danialfarid** [comment]
@rvanbaalen thanks for finding out the related code.
@jihoonkim87 The code was added to allow cross domain request on IE8/9.
Those browsers will not let you call xhr.open() for a cross domain url and it would throw Access is Denied error. The solution is to let that request go through by changing the url and then on xhr.send() it will be sent using flash which will work with CORS requests.

I fixed this at version 3.0.3 to throw the original error if the request is for not for this plugin's file upload.
